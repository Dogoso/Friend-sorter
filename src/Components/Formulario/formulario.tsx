import React, { useRef, useState } from "react"
import { Outlet } from "react-router-dom"
import useAdicionarParticipante from "State/Hooks/useAddParticipante"
import useGetErrorMessage from "State/Hooks/useGetErrorMessage"
import styles from "Styles/_globals.module.css"

const Formulario = () => {

    const [participante, setParticipante] = useState("")
    const participanteInputRef = useRef<HTMLInputElement>(null)
    const addParticipante = useAdicionarParticipante()
    const errorMessage = useGetErrorMessage()

    const participanteFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        addParticipante(participante)
        setParticipante("")
        participanteInputRef.current?.focus()
    }

    return (
        <section>
            {errorMessage && 
                <p role="errorMessage" className={styles.error}>
                    {errorMessage}
                </p>
            }
            <h1 className={styles.title}>
                Vamos come√ßar!
            </h1>
            <form onSubmit={participanteFormSubmit} className={styles.form}>
                <input 
                    ref={participanteInputRef}
                    className={styles.input}
                    type="text" 
                    placeholder="Insira os nomes dos participantes"
                    value={participante}
                    onChange={(e) => setParticipante(e.target.value)}
                />
                <button 
                    className={styles.add}
                    type="submit" 
                    disabled={!participante}
                >
                    Adicionar
                </button>
            </form>
            <div>
                <Outlet/>
            </div>
        </section>
    )
}

export default Formulario