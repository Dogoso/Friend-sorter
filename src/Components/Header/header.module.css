.header {
    background-color: #4B69FD;
    display: flex;
    padding: 50px 50px 0px 50px;
    justify-content: center;
}

.margin {
    margin-top: 50px;
}

.participante {
    position: relative;
    top: 35px;
}
@media only screen and (max-width: 767px) {
    .img {
        width: 100%;
    }
    .bg {
        display: flex;
        justify-content: center;
        flex-direction: column;
        align-items: center;
    }
    .participante {
        top: 30px;
    }
}