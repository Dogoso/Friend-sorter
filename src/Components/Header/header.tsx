import styles from "./header.module.css"

const Header = () => {
    return (
        <header className={styles.bg}>
            <div className={styles.margin}>
                <img 
                    src="src/imgs/logo.png" 
                    alt="Logo escrito sorteador de amigo secreto" 
                    className={styles.img}
                />
            </div>
            <div className={styles.participante}>
                <img 
                    src="src/imgs/participante.png" 
                    alt="Person playing shits"
                    className={styles.img}
                />
            </div>
        </header>
    )
}

export default Header