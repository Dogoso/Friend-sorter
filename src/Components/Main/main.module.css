.container {
    background-color: #FFF9EB;
    border: 2px solid black;
    border-top-left-radius: 60px;
    border-top-right-radius: 60px;
    min-height: calc(100vh + 60px);
    padding: 90px 50px 50px 50px;
    position: relative;
    top: -60px;
}