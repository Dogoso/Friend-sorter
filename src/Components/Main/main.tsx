import styles from "./main.module.css"

type Props = {
    children: JSX.Element,
};

const Main =  ({ children }: Props) => {
    return (
        <main className={styles.container}>
            {children}
        </main>
    )
}

export default Main