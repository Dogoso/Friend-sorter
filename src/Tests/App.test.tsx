import { render } from "@testing-library/react"
import Formulario from "Components/Formulario/formulario"
import Header from "Components/Header/header"
import Main from "Components/Main/main"
import React from "react"
import { RecoilRoot } from "recoil"

describe("Renderização da página principal", () => {

    test("Quando acessada", () => {
        const { container } = render(
            <RecoilRoot>
                <Header/>
                <Main>
                    <Formulario/>
                </Main>
            </RecoilRoot>
        )
        expect(container).toMatchSnapshot()
    })

})