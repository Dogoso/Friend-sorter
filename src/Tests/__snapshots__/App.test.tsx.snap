// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização da página principal Quando acessada 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="margin"
    >
      <img
        alt="Logo escrito sorteador de amigo secreto"
        class="img"
        src="src/imgs/logo.png"
      />
    </div>
    <div
      class="participante"
    >
      <img
        alt="Person playing shits"
        class="img"
        src="src/imgs/participante.png"
      />
    </div>
  </header>
  <main
    class="container"
  >
    <section>
      <h1
        class="title"
      >
        Vamos começar!
      </h1>
      <form
        class="form"
      >
        <input
          class="input"
          placeholder="Insira os nomes dos participantes"
          type="text"
          value=""
        />
        <button
          class="add"
          disabled=""
          type="submit"
        >
          Adicionar
        </button>
      </form>
      <div />
    </section>
  </main>
</div>
`;
